packages/components/src/DateTimePickers/LegacyDateTimePickers/DateTime/date-extraction.js

```diff
@@ -487,32 +543,47 @@ function extractPartsFromTextInput(textInput, options) {
 	let date;
 	let errors = [];
 	let dateTextToParse = textInput;
+	let dateMatched = false;
 
 	try {
 		if (options.useTime) {
 			const splitMatches = textInput.match(splitDateAndTimePartsRegex) || [];
 			if (!splitMatches.length) {
-				throw new DatePickerException('DATETIME_INVALID_FORMAT', 'DATETIME_INVALID_FORMAT');
+				if (!hybridMode) {
+					throw new DatePickerException('DATETIME_INVALID_FORMAT', 'DATETIME_INVALID_FORMAT');
+				}
 			} else {
+				dateMatched = true;
 				// extract date part from datetime
 				dateTextToParse = splitMatches[1];
 
 				// extract time part and parse it
 				try {
 					const timeTextToParse = splitMatches[2];
 					time = strToTime(timeTextToParse, options.useSeconds);
-					checkTime(time);
+					checkTime(time, hybridMode);
 				} catch (error) {
 					errors = errors.concat(error);
 				}
 			}
 		}
 
-		// parse date
-		try {
-			date = strToDate(dateTextToParse, options.dateFormat);
-		} catch (error) {
-			errors = errors.concat(error);
+		if (hybridMode && !dateMatched) {
+			// parse date
+			const hybridDateTime = extractDateOrTimeHybridMode(textInput, options);
+			date = hybridDateTime.date;
+			time = hybridDateTime.time;
+			if (!time) {
+				time = { hours: '', minutes: '', seconds: '' };
+			}
+			errors = errors.concat(hybridDateTime.errors);
+		} else {
+			// parse date
+			try {
+				date = strToDate(dateTextToParse, options.dateFormat);
+			} catch (error) {
+				errors = errors.concat(error);
+			}
```

Comment:
Instead of having a `dateMatched ` boolean to check if we executed a part of the code, move this code in l.554.

```
if (!splitMatches.length) {
  if(hybridMode) {
     // your code
  } else {
    throw new DatePickerException('DATETIME_INVALID_FORMAT', 'DATETIME_INVALID_FORMAT');
  }
} else {

}
```

Reply:
It can't works, you only solve the case when we have date OR time in `hybridMode`.
We need to passed in the `else` (where `dateMatched` is set to `true`) in cas we have date AND time, even in `hybridMode`.

 But i'm agreed this `dateMatched` is not the most elegant thing. I try some refactoring to avoid that, but it means change all this function body. I find it risky on this legacy component.